### PR TITLE
Fix an assortment of embedded Tomcat issues

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Changes
* Avoid session deserialization outside of LK webapp
* Fix additional webapp deployments not serving files
* Support separate port for Spring Boot shutdown endpoint
* Don't use Spring Boot's error pages

### CODE REVIEW

File: server/embedded/src/org/labkey/embedded/LabKeyTomcatServletWebServerFactory.java [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+        // Prevent the Spring Boot webapp from trying to deserialize the LabKey sessions
+        getSession().setPersistent(false);
+
+        // Don't use Spring Boot's error pages, as we want to render our own
+        setErrorPages(Collections.emptySet());
+
+        super.prepareContext(host, initializers);
+    }
+
+//    @Override
```
Comment (Sigmonia) [inline server/embedded/src/org/labkey/embedded/LabKeyTomcatServletWebServerFactory.java]: Any reason to keep this?

**Review by Sigmonia — APPROVED** [review]

**Review by labkey-tchad — APPROVED** [review]

File: server/embedded/src/org/labkey/embedded/LabKeyTomcatServletWebServerFactory.java [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+                        throw new IOException("Failed to create directory " + dir + " - please check file system permissions");
+                    }
+                }
+                zipIn.closeEntry();
+                entry = zipIn.getNextEntry();
+            }
+        }
+    }
+
+    private static final int BUFFER_SIZE = 4096;
```
Comment (Sigmonia) [inline server/embedded/src/org/labkey/embedded/LabKeyTomcatServletWebServerFactory.java]: Should probably move this up to top of the page

File: server/embedded/src/org/labkey/embedded/LabKeyTomcatServletWebServerFactory.java (hunk 1 above)
Comment (labkey-jeckels) [reply server/embedded/src/org/labkey/embedded/LabKeyTomcatServletWebServerFactory.java]: Nope. Deleted

File: server/embedded/src/org/labkey/embedded/LabKeyTomcatServletWebServerFactory.java (hunk 2 above)
Comment (labkey-jeckels) [reply server/embedded/src/org/labkey/embedded/LabKeyTomcatServletWebServerFactory.java]: I moved it. I sometimes find it helpful to keep the declaration close to the usage, but also agree that sprinkling them throughout the class isn't a good option

**Pull request merged**